src/collocateElement.utility.js

```diff
@@ -536,32 +560,70 @@ const getRequestedCollocation = (modifiers = {}, defaultPosition = 'bottom', def
   }
 };
 
+const touchCollocateElemeAt = (element, elementContent, trigger, arrow) => {
+  resetAndHideElement(element);
+  resetElementContentHeight(elementContent);
+  const elementRect = element.getBoundingClientRect();
+  const triggerRect = trigger.getBoundingClientRect();
+  const rect = {
+    elementRect,
+    triggerRect,
+    arrowRect: arrow,
+    contentExtraHeight: getElementContentExtraHeight(elementContent, elementRect.height),
+  };
+  const touchScreensElementCoordinates = mapCoordinatesToCenterPosition(rect, getViePort());
+  applyTouchScreensCoordinates(element, elementContent, { ...touchScreensElementCoordinates });
+};
+
+/* New version of IOS does not trigger any event when the vieport changes due to soft keyboar
+ * so we need to use an interval to detect changes.
+ * onVieportChangeInterval should be dismissed when the dorpdown is closed */
+const touchDetectVieportChangesAndCollocate = (element, elementContent, trigger, arrow) => {
+  const getState = () => ({
+    winHeight: window.innerHeight,
+    visualHeight: window.visualViewport?.height || 0,
+    visualOffsetTop: window.visualViewport?.offsetTop || 0,
+  });
+  const isSameEstate = (e1, e2) => (
```

Comment:
Maybe word `estate` can be confusing with parameters like `e1` and `e2`. What do you think about using something like:
```suggestion
  const isSameDropdownPosition = (oldPosition, newPosition) => (
```

or
```suggestion
  const isSameState = (oldState, newState) => (
```

Reply:
the state is related to the window dimensions rather than the dropdown.  so maybe `windowDimensions` would be a better fit?

Reply:
yeah, sounds perfect mate 👍 👍